lib/public/views/Runs/format/formatRunDetectors.js

```diff
@@ -29,23 +29,51 @@ export const formatRunDetectors = (run, detailsModel) => {
         return '-';
     }
 
+    const _detectorTypes = detailsModel.detectorTypes.match({
+        NotAsked: () => [],
+        Loading: () => [],
+        Success: (data) => data,
+        Failure: (error) => error,
+    });
+
+    const _allAndSelectedDetectorTypes = _detectorTypes.map(({ id, name }) => {
+        const isPresent = detectorsQualities.map((obj) => obj.id).includes(id);
+        const quality = isPresent ? detectorsQualities.filter((obj) => obj.id == id)[0].quality : null;
+        return { id, name, quality, isPresent };
+    });
+
     const detectorsBadges = h(
         '.g2.flex-row.flex-grow.flex-wrap.justify-end',
-        detectorsQualities.map(({ id, name, quality }) => {
+        _allAndSelectedDetectorTypes.map(({ id, name, quality, isPresent }) => {
             const actualQuality = detailsModel.isEditModeEnabled
                 ? detailsModel.runPatch.getDetectorQuality(id) || quality
                 : quality;
-            const qualityIsBad = actualQuality === RunDetectorQualities.BAD;
-            const color = qualityIsBad ? 'danger' : 'success';
+            let color;
+            let icon;
+            switch (actualQuality) {
+                case RunDetectorQualities.GOOD:
+                    color = 'success';
+                    icon = iconCheck();
+                    break;
+                case RunDetectorQualities.BAD:
+                    color = 'danger';
+                    icon = iconX();
+                    break;
+                default:
+                    color = 'gray';
+                    icon = iconBan();
+                    break;
+            }
             return h(
-                `span.detector-badge.flex-row.justify-center.lh1.g1.br2.pv1.ph2.b1.b-${color}.${color}`,
+                `span.detector-badge.flex-row.justify-center.lh1.g1.br2.pv1.ph2.b1.b-${color}.${color}${isPresent ? '.isPresent' : ''}`,
```

Comment:
I think that you should not create classes specifically for tests (it's a mistake commonly done in bookkeeping code), test should never impact the actual code.
Instead in the tests, you know from the seeders what detectors and qualities to expect, you can then check the color that is applied.